packages/typespec-test/test/batch_modular/generated/typespec-ts/src/util/serializeUtil.ts

```diff
@@ -0,0 +1,3372 @@
+  BatchTaskListSubtasksResultOutput as RestBatchTaskListSubtasksResult,
+  TaskAddResultOutput as RestTaskAddResult,
+  TaskAddCollectionResultOutput as RestTaskAddCollectionResult,
+  TaskStatisticsOutput as RestTaskStatistics,
+  BatchNodeInformationOutput as RestBatchNodeInformation,
+  BatchTaskOutput as RestBatchTask,
+  BatchTaskListResultOutput as RestBatchTaskListResult,
+  FilePropertiesOutput as RestFileProperties,
+  NodeFileOutput as RestNodeFile,
+  NodeFileListResultOutput as RestNodeFileListResult,
+  NodeVMExtensionListOutput as RestNodeVMExtensionList,
+  InstanceViewStatusOutput as RestInstanceViewStatus,
+  VMExtensionInstanceViewOutput as RestVMExtensionInstanceView,
+  VMExtensionOutput as RestVMExtension,
+  NodeVMExtensionOutput as RestNodeVMExtension,
+  BatchNodeListResultOutput as RestBatchNodeListResult,
+  UploadBatchServiceLogsResultOutput as RestUploadBatchServiceLogsResult,
+  BatchNodeRemoteLoginSettingsResultOutput as RestBatchNodeRemoteLoginSettingsResult,
+  ImageReferenceOutput as RestImageReference,
+  VirtualMachineInfoOutput as RestVirtualMachineInfo,
+  NodeAgentInformationOutput as RestNodeAgentInformation,
+  InboundEndpointOutput as RestInboundEndpoint,
+  BatchNodeEndpointConfigurationOutput as RestBatchNodeEndpointConfiguration,
+  BatchNodeErrorOutput as RestBatchNodeError,
+  CertificateReferenceOutput as RestCertificateReference,
+  StartTaskInformationOutput as RestStartTaskInformation,
+  AutoUserSpecificationOutput as RestAutoUserSpecification,
+  UserIdentityOutput as RestUserIdentity,
+  EnvironmentSettingOutput as RestEnvironmentSetting,
+  ResourceFileOutput as RestResourceFile,
+  BatchNodeIdentityReferenceOutput as RestBatchNodeIdentityReference,
+  ContainerRegistryOutput as RestContainerRegistry,
+  TaskContainerSettingsOutput as RestTaskContainerSettings,
+  StartTaskOutput as RestStartTask,
+  NameValuePairOutput as RestNameValuePair,
+  TaskFailureInformationOutput as RestTaskFailureInformation,
+  TaskContainerExecutionInformationOutput as RestTaskContainerExecutionInformation,
+  TaskExecutionInformationOutput as RestTaskExecutionInformation,
+  TaskInformationOutput as RestTaskInformation,
+  BatchNodeOutput as RestBatchNode,
+  BatchErrorDetailOutput as RestBatchErrorDetail,
+  ErrorMessageOutput as RestErrorMessage,
+  BatchErrorOutput as RestBatchError,
+} from "../rest/index.js";
+
+export function serializeNodeRemoveOptions(
+  o: NodeRemoveOptions,
+): RestNodeRemoveOptions {
+  return {
+    nodeDeallocationOption: o["nodeDeallocationOption"],
+    resizeTimeout:
+      o["resizeTimeout"] === undefined ? o["resizeTimeout"] : FIXME,
+    nodeList: o["nodeList"],
+  };
+}
+
+export function serializeBatchPoolReplaceOptions(
+  o: BatchPoolReplaceOptions,
+): RestBatchPoolReplaceOptions {
+  return {
+    targetNodeCommunicationMode: o["targetNodeCommunicationMode"],
+    metadata: o["metadata"].map((e) => MISSING_SERIALIZER(e)),
```

Comment:
Is this because we haven't started to handle array element type yet? Also, when we are handling the serialization inside one model, we could assume other model's serialization models has been build somewhere else, so what we need here is just that serializer's name. for example, if metadata's element type is called MetadataElement, we can just generate 
```suggestion
    metadata: o["metadata"].map((e) => serializeMetadataElement(e)),
```